katdal/dataset.py

```diff
@@ -190,6 +210,46 @@ def __hash__(self):
         """Base hash on description tuple, just like equality operator."""
         return hash(self._description)
 
+    def subset(self, first, last):
```

Comment:
I prefer the name `slice` or `subrange` as `subset` seems too generic, but my conviction is relatively weak on this.

Reply:
I was also not totally happy with subset. I considered slice, but I dislike functions that have the same name as a built-in type (if only because my editor gives misleading syntax highlighting in that case). Subrange sounds reasonable. Another option would be to allow `__getitem__` e.g. spw[first:last].

Reply:
I also like `__getitem__`, as long as strides raise an exception (in case users expect more than we can provide :-))

Reply:
I started implementing `__getitem__`, but then realised that might lead to expectations that one can do things like `spw[:-2]` which would complicate the implementation, so I'm going with subrange.